src/components/MultiSelect/MultiSelect.test.tsx

```diff
@@ -23,42 +23,37 @@ const items = [
   { id: "furniture", name: "Furniture" },
 ];
 
-const MultiSelectTestDialogComponent = (componentId) => {
+const MultiSelectTestDialogComponent = ({ multiSelectId }) => {
   const { onChange, onMixedStateChange, selectedItems, onClear } =
-    useMultiSelect({
-      multiSelectId: componentId,
-      items,
-    });
+    useMultiSelect();
   return (
     <MultiSelect
-      id={componentId}
+      id={multiSelectId}
       label="MultiSelect Label"
       variant="dialog"
       items={items}
       selectedItems={selectedItems}
-      onChange={(e) => onChange(e.target.id)}
+      onChange={(e) => onChange(e.target.id, multiSelectId)}
       onMixedStateChange={(e) => {
-        onMixedStateChange(e.target.id);
+        onMixedStateChange(e.target.id, multiSelectId, items);
       }}
-      onClear={() => onClear()}
+      onClear={() => onClear(multiSelectId)}
       onApply={() => null}
     />
   );
 };
-const MultiSelectTestListboxComponent = (componentId) => {
-  const { onChange, selectedItems, onClear } = useMultiSelect({
-    multiSelectId: componentId,
-    items,
-  });
+const MultiSelectTestListboxComponent = ({ multiSelectId }) => {
+  const { onChange, selectedItems, onClear } = useMultiSelect();
   return (
     <MultiSelect
-      id={componentId}
+      id={multiSelectId}
       label="MultiSelect Label"
       variant="listbox"
       items={items}
       selectedItems={selectedItems}
-      onChange={(selectedItem) => onChange(selectedItem.id)}
-      onClear={() => onClear()}
+      onChange={(e) => onChange(e.id, multiSelectId)}
```

Comment:
`selectedItem` is a better variable name than `e` since `e` is typically short for "event" but that's not the case for the "listbox" variant's `onChange` callback.